### PR TITLE
TASK-53494: change  creating challenge permissions to administrators

### PR DESCRIPTION
**Pull request by GouadriaHanen**
change create challenge permission to administrators:
use external configuration InitParams of the service and the possibility to fill it by a value from exo.properties.

### CODE REVIEW

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java [hunk 1]
```diff
@@ -95,8 +96,8 @@ public Challenge updateChallenge(Challenge challenge, String username) throws Il
   }
 
   @Override
-  public boolean canAddChallenge(String currentUser) throws Exception {
-    return spaceService.getManagerSpaces(currentUser).getSize() > 0;
+  public boolean canAddChallenge(String currentUser)  {
+    return ConversationState.getCurrent().getIdentity().isMemberOf("/platform/administrators");
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java]: This should be configured with an external configuration in the InitParams of the service and possibly filled by a value from exo.properties.

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java (hunk 1 above)
Comment (GouadriaHanen) [reply services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java]: done!

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java [hunk 2]
```diff
@@ -22,9 +24,16 @@
...
-  public ChallengeServiceImpl(ChallengeStorage challengeStorage, SpaceService spaceService) {
+  private String     CREATORS_GROUP;
+
+  private static final String     CREATORS_GROUP_KEY             = "challenge.creator.group";
+
+
+  public ChallengeServiceImpl(ChallengeStorage challengeStorage, SpaceService spaceService, InitParams params) {
     this.challengeStorage = challengeStorage;
     this.spaceService = spaceService;
+    this.CREATORS_GROUP = params.getValueParam(CREATORS_GROUP_KEY).getValue();
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java]: Please check for nullity here
```suggestion
    this.groupOfCreators = params.getValueParam(CREATORS_GROUP_KEY).getValue();
```

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java [hunk 3]
```diff
@@ -95,8 +104,8 @@ public Challenge updateChallenge(Challenge challenge, String username) throws Il
   }
 
   @Override
-  public boolean canAddChallenge(String currentUser) throws Exception {
-    return spaceService.getManagerSpaces(currentUser).getSize() > 0;
+  public boolean canAddChallenge()  {
+    return ConversationState.getCurrent().getIdentity().isMemberOf(CREATORS_GROUP);
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java]: Could you add the Check for nullity

**Review by ahamdi — CHANGES_REQUESTED** [review]

**Comment by ahamdi** [comment]
Update the PR description

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java [hunk 4]
```diff
@@ -95,8 +105,11 @@ public Challenge updateChallenge(Challenge challenge, String username) throws Il
   }
 
   @Override
-  public boolean canAddChallenge(String currentUser) throws Exception {
-    return spaceService.getManagerSpaces(currentUser).getSize() > 0;
+  public boolean canAddChallenge()  {
+    if (groupOfCreators != null){
+      return ConversationState.getCurrent().getIdentity().isMemberOf(groupOfCreators);
+    }
+    else return false;
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java]:
```suggestion
    return false;
```

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java [hunk 5]
```diff
@@ -95,8 +105,11 @@ public Challenge updateChallenge(Challenge challenge, String username) throws Il
   }
 
   @Override
-  public boolean canAddChallenge(String currentUser) throws Exception {
-    return spaceService.getManagerSpaces(currentUser).getSize() > 0;
+  public boolean canAddChallenge()  {
+    if (groupOfCreators != null){
+      return ConversationState.getCurrent().getIdentity().isMemberOf(groupOfCreators);
+    }
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java:111]:
```suggestion
    if (groupOfCreators != null && ConversationState.getCurrent() != null && ConversationState.getCurrent().getIdentity() != null){
      return ConversationState.getCurrent().getIdentity().isMemberOf(groupOfCreators);
    }
```

**Review by ahamdi — APPROVED** [review]

**Pull request merged**